api/routes/api_login.py

```diff
@@ -99,4 +99,17 @@ async def log_in(service: UserService = Provide[Container.user_service]):
 
     except CustomQSportException as e:
         logger.error(f"Error in POST /login: {str(e)}")
-        return get_custom_error_response(e)
\ No newline at end of file
+        return get_custom_error_response(e)
+
+@login_app.route("/refresh", methods=['POST'])
+@inject
+@handle_exceptions
+@logger.log_function_call()
+@jwt_required(refresh=True)
+async def refresh(service: UserService = Provide[Container.user_service]):
+    try:
+        return await service.refresh_tokens()
+    except CustomQSportException as e:
+        logger.error(f"Error in POST /refresh: {str(e)}")
+        
+
```

Comment:
Redundant empty lines are still present. You should've removed them 2 weeks ago :)